lib/charms/vault_k8s/v0/vault_kv.py

```diff
@@ -519,15 +533,19 @@ def __init__(
         charm: ops.CharmBase,
         relation_name: str,
         mount_suffix: str,
+        refresh_events: Optional[List] = None,
     ) -> None:
         super().__init__(charm, relation_name)
         self.charm = charm
         self.relation_name = relation_name
         self.mount_suffix = mount_suffix
+        self._refresh_events = refresh_events or [self.charm.on.config_changed]
```

Comment:
I'm not sure I understand why we observe config changed as a default? Nothing here depends on the requiring charm config right? We may want to use `update status` instead.

Reply:
This change is initiated by a unit going down then coming up again with a new IP address.
Based on the [documentation](https://juju.is/docs/sdk/config-changed-event) of the `config-changed` event it seems to me that a unit changing its IP will trigger a `config-changed` event.

Reply:
How do you imagine people would use `refresh_events`? Isn't it enough to just look at `config_changed` since that fires when the IP changes?

Reply:
Users don't need to use `refresh_event`, they can specify one depending on their use case. And if they don't we jut use the `config_changed` event.

Reply:
I'm wondering if they will ever need or want to specify one. What use case is there for it? can you give an example?

Reply:
No, I don't have a use case on top of my mind.
We can silently do `config-changed` in the background if we are convinced that users won't change it.
But I don't mind keeping the flexibility for example if a user doesn't want to trigger this handler on each config change.
what do you think @Kayra1 ?

Reply:
I think we should remove this option entirely and do it silently with just `config-changed`. It's one less thing for users to worry about and there doesn't seem to be a catch with it that users need to code around or know about.

We can introduce it in the future if someone needs it, but I can't imagine why anyone would want to create a custom event just for one of their libraries either.

Reply:
Event removed.